File: eco-core/core-backend/src/main/java/com/willfp/eco/internal/logging/EcoLogger.java
package com.willfp.eco.internal.logging;

import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.util.StringUtils;
import org.jetbrains.annotations.NotNull;

import java.util.logging.Level;
import java.util.logging.Logger;

public class EcoLogger extends Logger {
    public EcoLogger(@NotNull final EcoPlugin plugin) {
        super(plugin.getName(), (String) null);
        String prefix = plugin.getDescription().getPrefix();
        this.setParent(plugin.getServer().getLogger());
        this.setLevel(Level.ALL);
    }

    @Override
    public void info(@NotNull final String msg) {
        super.info(StringUtils.translate(msg));
    }

    @Override
    public void warning(@NotNull final String msg) {
        super.warning(StringUtils.translate(msg));
    }

    @Override
    public void severe(@NotNull final String msg) {
        super.severe(StringUtils.translate(msg));
    }
}

File: eco-core/core-backend/src/main/kotlin/com/willfp/eco/internal/logging/EcoLogger.kt
package com.willfp.eco.internal.logging

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.util.StringUtils
import java.util.logging.Level
import java.util.logging.Logger
import org.bukkit.Bukkit

class EcoLogger(private val plugin: EcoPlugin) : Logger(plugin.name, null as String?) {
    override fun info(msg: String) {
        Bukkit.getConsoleSender().sendMessage("[${plugin.name}] ${StringUtils.format(msg, StringUtils.FormatOption.WITHOUT_PLACEHOLDERS)}")
    }

    init {
        parent = plugin.server.logger
        this.level = Level.ALL
    }
}
